x/evm/keeper/state_transition.go

```diff
@@ -519,15 +519,22 @@ func (k *Keeper) ResetGasMeterAndConsumeGas(ctx sdk.Context, gasUsed uint64) {
 	ctx.GasMeter().ConsumeGas(gasUsed, "apply evm transaction")
 }
 
+// GetProposerAddress returns current block proposer's address when provided proposer address is empty.
+func GetProposerAddress(ctx sdk.Context, proposerAddress []byte) []byte {
```

Comment:
let's use the consensus address whenever possible
```suggestion
func GetProposerAddress(ctx sdk.Context, proposerAddress []byte) sdk.ConsAddress {
```